framework/src/st_things/things_stack/utils/things_network.c

```diff
@@ -380,12 +381,41 @@ void things_wifi_scan_done(wifi_manager_scan_info_s **scan_result, int res)
 	access_point_info_s *pinfo = NULL;
 	access_point_info_s *p_last_info = NULL;
 	while (wifi_scan_iter != NULL) {
-		if (wifi_scan_iter->ssid != NULL) {
+		if (strlen(wifi_scan_iter->ssid) != 0) {
```

Comment:
This might result in segmentation fault if "wifi_scan_iter->ssid" is NULL.
Change it as:-
if (wifi_scan_iter->ssid != NULL && strlen(wifi_scan_iter->ssid) != 0)